source/fe/fe_values_base.cc

```diff
@@ -152,9 +165,17 @@ FEValuesBase<dim, spacedim>::CellIteratorContainer::n_dofs_for_dof_handler()
   const
 {
   Assert(is_initialized(), ExcNotReinited());
-  Assert(dof_handler != nullptr, ExcNeedsDoFHandler());
 
-  return dof_handler->n_dofs();
+  switch (cell.value().index())
+    {
+      case 1:
+        return std::get<1>(cell.value())->get_dof_handler().n_dofs();
+      case 2:
+        return std::get<2>(cell.value())->get_dof_handler().n_dofs();
```

Comment:
Unrelated to this patch, but shouldn't we return
```suggestion
        return std::get<2>(cell.value())->get_dof_handler().n_dofs(std::get<2>(cell.value())->level());
```
i.e., the number of level dofs rather than the number of dofs globally?

Reply:
That's not what the code originally did, nor what is apparently necessary to pass the test suite. This function is used in contexts such as this:
```
template <int dim, int spacedim>
template <typename Number>
void
FEValuesBase<dim, spacedim>::get_function_values(
  const ReadVector<Number> &fe_function,
  std::vector<Number>      &values) const
{
  Assert(this->update_flags & update_values,
         ExcAccessToUninitializedField("update_values"));
  AssertDimension(fe->n_components(), 1);
  Assert(present_cell.is_initialized(), ExcNotReinited());
  AssertDimension(fe_function.size(), present_cell.n_dofs_for_dof_handler());

  // get function values of dofs on this cell
  Vector<Number> dof_values(dofs_per_cell);
  present_cell.get_interpolated_dof_values(fe_function, dof_values);
```
We pass in a global vector and interpolate it onto the current cell (active or not).

Reply:
In other words, I believe that the change I made is correct.

Reply:
Thanks for checking, this is not what I expected to happen, but I do see it now.